src/PAModel/PAConvert/Yaml/YamlWriter.cs

```diff
@@ -110,6 +110,13 @@ public void WriteProperty(string propertyName, string value)
                         WriteIndent();
                         needIndent = false;
                     }
+
+                    if (ch == '\r')
```

Comment:
\r shouldn't be making it to the yamlwriter, see a couple places in IRStateHelper where we do something like  
`           var script = rule.InvariantScript.Replace("\r\n", "\n").Replace("\r", "\n").TrimStart();`

What kind of property is hitting this and why is `\r\n` making its way into the IR to be written?

Reply:
Maybe the right thing to do here is to define an actual constructor for ExpressionNode and do this trimming/normalizing there.

Reply:
The yaml layer should handle \r  (we can't rely on the upstream caller to remove them).  Ie, see new test added in this pr. 
In this bug, there's probably some path that's skipping the IR's replacement.

Reply:
We're definitely missing a bunch of spots in the IRStateHelper. 
Although if we do it in the lexer, then we should be able to completely remove the conversions from IRStateHelper . I'll try that ...